### DISCUSSION TITLE
[2024-04-29] Library & Engine Team Update

### DISCUSSION THREAD

**Discussion by GregoryTravis** [enso-org/enso#9803] [Weekly Demo Day]
- James: Working through various issues from OKR / blog use. #9739
- James: `Excel_Workbook.read_many` #9759
- James: Improve Table Viz handling of Numbers #9793
- Radosław: Files (mainly Cloud)
	- Writing Cloud files #9686
	- Implement `recursive` argument for `File.delete` #9719
	- Writing to Data Links #9750
    - Fix `copy_to`/`move_to` edge case when source and target are the same #9798
- Radosław: Improving UX of Test suite
	- Highlight stack trace elements from the current project in Test suite reports #9724
	- Print pending groups in-line and include their reason, also add progress bar to the runner #9796
- Radosław: Fix `Json.parse ""` edge case and add a test for it #9722
- Radosław: Quickfix adding missing dependency in `Standard.AWS` library + improvements to the UX of the license review tool #9782 
- Radosław: Types
	- Planning next steps after PoC 1
	- Will start by working on resolving types for method calls
		- First step: adding another small pass that will allow to easily benefit from type signatures of methods
	- Later: default arguments handling
- Greg: On MacOS, do not assume existence of `Contents/Home` in the JDK binary path #9684
- Greg: Add `Decimal.floor`, `.ceil`, and `.trunc` #9694
- Greg: `Integer.to_decimal` and `Float.to_decimal` #9716
- Greg: Add tests for BigDecimal internal representation methods #9717
- Greg: Change ‘decimal’ to ‘float’ in Numbers_Spec #9718
- Greg: Implement `Vector.build`: implementation and some tests #9725
- Greg: Convert `Vector.new_builder` to `Vector.build` in `Standard.Base` #9754
- Greg: Add `Vector.build_multiple` and handle dataflow error and warning propagation in `build` and `build_multiple` #9766
- Cassandra: Learning new community platform
- Cassandra: Reviewed table and column components for consistency
- Cassandra: Worked on real-world university use case for Enso
- Ad: Make errors propagate out of each #9742
- Ad: Make excel writer work for custom types #9752
- Ad: Add new icons #9756
- Ad: Add Ignored_Nothing_Values #9770
- Ad: Make Table.should_equal and Column.should_equal consider NaN equal #9799
- Jaroslav: Make conversion of double to BigDecimal exact #9740
- Jaroslav: Allow ..Nothing to resolve to Nothing #9746
- Jaroslav: Special treatment for ..True and ..False #9767
- Jaroslav: Use InteropLibrary when accepting String values #9773
- Jaroslav: Removing Truffle API dependency from `runtime-compiler` project #9785
- Jaroslav: `Table.join`: #9410
- Hubert: Experiment in lock-free module invalidation #9639
- Hubert: Open Java modules for Snowflake setup #9664
- Hubert: Fix bundled engine after GraalVM upgrade #9735
- Hubert: In-memory suggestions #9751
- Hubert: WIP: Initial Bazel integration #9777
- Pavel: Update to GraalVM 24.0.0 #9647
- Pavel: Export works without import #9683
- Pavel: Atom constructors can be private #9692
- Pavel: Panics can be caught from TCO loops #9705
- Pavel: [WIP] Add logging for UnresolvedMethod bug #9763
- Dmitry: Language Server API for AI #9679
- Dmitry: Should not use auto-scoping for atom types #9743
- Dmitry: Autoscoping should not escape True and False #9797

**Comment by radeusgd** [comment]
Test suite UX improvements:

https://github.com/enso-org/enso/assets/1436948/8815cbcd-8a68-41db-832f-7f357a0de681

![image](https://github.com/enso-org/enso/assets/1436948/b02da409-b67c-421c-a6d0-07a6cbbaf5f8)
